pkg/client/unversioned/helper.go

```diff
@@ -257,24 +257,31 @@ func NewOrDie(c *Config) *Client {
 // running inside a pod running on kuberenetes. It will return an error if
 // called from a process not running in a kubernetes environment.
 func InClusterConfig() (*Config, error) {
-	token, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/" + api.ServiceAccountTokenKey)
-	if err != nil {
-		return nil, err
-	}
-	tlsClientConfig := TLSClientConfig{}
-	rootCAFile := "/var/run/secrets/kubernetes.io/serviceaccount/" + api.ServiceAccountRootCAKey
-	if _, err := util.CertPoolFromFile(rootCAFile); err != nil {
-		glog.Errorf("Expected to load root CA config from %s, but got err: %v", rootCAFile, err)
+	config := &Config{}
+	if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount"); err != nil {
```

Comment:
@liggitt AFAIU we keep certificates in this directory (that's how I read old code). So without those there's no TLS, right?

@lavalamp I don't think we can assume that it's either 443 or 80. IIRC we just set 443 in our configuration - it's not hardcoded. I think user can choose arbitrary port number, right?